code/src/components/LoadingIndicator.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+import Loader from 'react-loader-spinner'
+
+export const LoadingIndicator = () => {
+  const isLoading = useSelector(state => state.ui.isLoadin)
+
+  return (
+    <>
+      {isLoading && (
+        <Loader
+          type="Puff"
+          color="#00BFFF"
+          height={100}
+          width={100}
+          timeout={3000} //3 secs
+        />
```

Comment:
Regarding props.
This is a component that we get from a little package called react-loader-spinner.
We don't know exactly how that component file looks like but in the documentation we saw that it could deal with us passing in props such as type, color, timeout and so on. This is because the developer creating that loader made it so. Then that file, that lives in another place that we dont need to care about takes whatever we pass in to type="Puff" for eaxmple and shows an image called puff in that little package.
This is an example on how to pass in props to make the component look or work in more dynamic ways. We don't need multiple loading indicators, just one that will show different images regarding what type it get passed in. 
(Let me know if this is still unclear!)